src/p2p/net_node.inl

```diff
@@ -1019,19 +1019,20 @@ namespace nodetool
               {
                   post_request_to_supernode<cryptonote::COMMAND_RPC_BROADCAST>("broadcast", arg, arg.callback_uri);
               }
+
+              int next_hop = arg.hop - 1;
```

Comment:
The logic here isn't new, but this is relying on undefined behaviour: `arg.hop` is an unsigned int, so `arg.hop - 1` is an unsigned operation; the cast from `unsigned` -> `int` for an underflowed value is undefined.

This (and the various other versions of this later on) could avoid this and be simplified if written as:

```C++
if (arg.hop > 0) {
    arg.hop--;
    LOG_PRINT_L0("P2P Request: handle_broadcast: notify peers " << arg.hop);
    ...
```